### PR TITLE
AZP/JUCX: Build and release JUCX in centos7 container.

### PR DESCRIPTION
**Pull request by petro-rudenko**
## What
Build and release JUCX in centos7 container.

## Why ?
In a release pipeline we build libjucx binary. We have to build it on the lowest supported glibc and gcc. 
@abellina pointed out that there's an issue that a new gcc compiler that is used in fedora by default builds c++ (libjucx uses c++) in c++11 standard, that's not ABI compatible if you have libcpp98.

### CODE REVIEW

**Comment by abellina** [comment]
@petro-rudenko thanks for looking at this.

Discussed with @jlowe and the main thing that we need to make sure of is that centos7 container (and toolchain) used is not getting upgraded, e.g. that it is just the default toolchain that comes with centos7.

From https://github.com/openucx/ucx/blob/master/buildlib/centos7.Dockerfile, it certainly looks like it's just installing the default gcc, and the centos:7 image seems to be the standard one. If this is what you are using to build libjucx, we are pretty confident this so should run anywhere.

File: bindings/java/src/main/native/Makefile.am [hunk 1]
```diff
@@ -62,7 +62,7 @@ libjucx_la_SOURCES = context.cc \
                      ucs_constants.cc \
                      worker.cc
 
-libjucx_la_CXXFLAGS = -fPIC -DPIC -Werror
+libjucx_la_CXXFLAGS = -fPIC -DPIC -Werror -std=c++98
```
Comment (yosefe) [inline bindings/java/src/main/native/Makefile.am:65]: why is this needed?

Comment (petro-rudenko) [reply bindings/java/src/main/native/Makefile.am:65]: Thought to explicitly force c++98, even when compiling on  new compilers.

Comment (petro-rudenko) [reply bindings/java/src/main/native/Makefile.am:65]: https://stackoverflow.com/a/51396583/3271168
```
For example, in RHEL 6 g++ (GCC) 4.4.7 20120313 (Red Hat 4.4.7-23), the output:

         gnu++98
           GNU dialect of -std=c++98.  This is the default for C++ code.
And in Fedora 28 g++ (GCC) 8.1.1 20180502 (Red Hat 8.1.1-1), the output:

       gnu++1y
           GNU dialect of -std=c++14.  This is the default for C++ code.  The name gnu++1y 
```

Comment (yosefe) [reply bindings/java/src/main/native/Makefile.am:65]: seems like a non-portable flag..

Comment (petro-rudenko) [reply bindings/java/src/main/native/Makefile.am:65]: `-std=gnu++98` supported by gcc, clang, icc.

Comment (yosefe) [reply bindings/java/src/main/native/Makefile.am:65]: okay
